### PR TITLE
Remove the memory ID for external memory blocks.

### PR DESCRIPTION
**Pull request by ulysseB**
This commit removes external arrays from the objects exposed to the
constraint satisfaction problem. This ends up simplifying or removing
useless code, which will simplify later changes.

### CODE REVIEW

File: src/ir/mem.rs [hunk 1]
```diff
@@ -5,67 +5,31 @@ use utils::*;
...
-}
-
 #[derive(Clone, Copy, PartialEq, Eq, Hash, Debug, Serialize, Deserialize)]
 #[repr(C)]
 /// cbindgen:field-names=[id]
-pub struct InternalId(pub u32);
+pub struct MemId(pub u32);
 
-impl Into<usize> for InternalId {
+impl Into<usize> for MemId {
```
Comment (nicoTolly) [inline src/ir/mem.rs]: According to the rust doc, implementing From should be preferred to implementing Into when possible. So, replace this by impl From<MemId> for usize

File: src/ir/mod.rs [hunk 2]
```diff
@@ -295,6 +293,15 @@ where
     }
 }
 
+impl<I, T> Default for SparseVec<I, T>
+where
```
Comment (nicoTolly) [inline src/ir/mod.rs]: This where seems useless in the context

Comment (ulysseB) [reply src/ir/mod.rs]: It is required by SparseVec.

**Comment by ulysseB** [comment]
bors retry

**Comment by ulysseB** [comment]
bors retry

**Comment by ulysseB** [comment]
bors retry

**Pull request merged**